apps/docs/components/ProjectConfigVariables/ProjectConfigVariables.ComboBox.tsx

```diff
@@ -0,0 +1,103 @@
+import { Check, ChevronsUpDown } from 'lucide-react'
+import { noop } from 'lodash'
+import { useState } from 'react'
+import {
+  Button_Shadcn_ as Button,
+  Popover_Shadcn_ as Popover,
+  PopoverTrigger_Shadcn_ as PopoverTrigger,
+  PopoverContent_Shadcn_ as PopoverContent,
+  Command_Shadcn_ as Command,
+  CommandInput_Shadcn_ as CommandInput,
+  CommandEmpty_Shadcn_ as CommandEmpty,
+  CommandItem_Shadcn_ as CommandItem,
+  CommandGroup_Shadcn_ as CommandGroup,
+  cn,
+  ScrollArea,
+} from 'ui'
+
+export interface ComboBoxOption {
+  id: string
+  value: string
+  displayName: string
+}
+
+export function ComboBox<Opt extends ComboBoxOption>({
+  isLoading,
+  disabled,
+  name,
+  options,
+  selectedOption,
+  onSelectOption = noop,
+  className,
+}: {
+  isLoading: boolean
+  disabled?: boolean
+  name: string
+  options: Opt[]
+  selectedOption?: string
+  onSelectOption?: (newValue: string) => void
+  className?: string
+}) {
+  const [open, setOpen] = useState(false)
+
+  const selectedOptionDisplayName = options.find(
+    (option) => option.value === selectedOption
+  )?.displayName
+
+  return (
+    <Popover open={open} onOpenChange={setOpen}>
+      <div className={className}>
+        <PopoverTrigger asChild aria-label={`Select ${name}`} disabled={disabled}>
+          <Button
+            variant="outline"
+            role="combobox"
+            aria-expanded={open}
+            className={cn(
+              'overflow-hidden',
+              'h-auto min-h-10',
+              'flex justify-between',
+              'border-none',
+              'py-0 pl-0 pr-1 text-left'
+            )}
+          >
+            {isLoading
+              ? 'Loading...'
+              : options.length === 0
+              ? `No ${name} found`
+              : selectedOptionDisplayName ?? `Select a ${name}...`}
+            <ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />
+          </Button>
+        </PopoverTrigger>
+      </div>
+      <PopoverContent className="z-[999999] p-0" side="bottom">
```

Comment:
you shouldn't need to have `z-[999999]` 

something is wrong

Reply:
😂 was this resolved by mistake?

Reply:
nope should be gone now! 
![image](https://github.com/supabase/supabase/assets/26616127/8413cc88-643b-4948-ac96-c5347c042690)